sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.submitApplication.e2e-spec.ts

```diff
@@ -663,6 +664,187 @@ describe("ApplicationStudentsController(e2e)-submitApplication", () => {
+      "the E2 restriction code BUT the student already has an active RB restriction.",
+    async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
+
+      // Add an RB restriction to the student before submitting the application
+      const rbRestriction = await db.restriction.findOne({
+        where: { restrictionCode: RestrictionCode.RB },
+      });
+      await db.studentRestriction.save({
+        student,
+        restriction: rbRestriction,
+        isActive: true,
+      });
+
+      // Create a draft application with E2 restriction in payload
+      const draftApplication = await saveFakeApplication(
+        db.dataSource,
+        { student, programYear: recentActiveProgramYear },
+        {
+          applicationData: {} as ApplicationData,
+          offeringIntensity: OfferingIntensity.partTime,
+        },
+      );
+      const selectedOffering = await db.educationProgramOffering.save(
+        createFakeEducationProgramOffering(
+          {
+            auditUser: student.user,
+            institutionLocation: draftApplication.location,
+          },
+          {
+            initialValues: {
+              studyStartDate: getISODateOnlyString(
+                addDays(1, recentActiveProgramYear.startDate),
+              ),
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
+      const selectedProgram = selectedOffering.educationProgram;
+      const applicationData = {
+        selectedOfferingDate: selectedOffering.studyStartDate,
+        selectedOfferingEndDate: selectedOffering.studyEndDate,
+        selectedProgram: selectedProgram.id,
+        selectedOffering: selectedOffering.id,
+        selectedLocation: selectedOffering.institutionLocation.id,
+        restrictions: [RestrictionCode.E2],
+      };
+      const payload = {
+        associatedFiles: [],
+        data: applicationData,
+        programYearId: recentActiveProgramYear.id,
+      } as SaveApplicationAPIInDTO;
+      const endpoint = `/students/application/${draftApplication.id}/submit`;
+      const token = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+      const dryRunSubmissionMock = jest.fn().mockResolvedValue({
+        valid: true,
+        formName: FormNames.Application,
+        data: { data: applicationData },
+      });
+      formService.dryRunSubmission = dryRunSubmissionMock;
+      await mockUserLoginInfo(appModule, student);
+
+      // Act
+      await request(app.getHttpServer())
+        .patch(endpoint)
+        .send(payload)
+        .auth(token, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect({});
+
+      // Assert
+      // Verify that no new E2 restriction was added. The student should still only have the RB restriction.
+      const studentRestrictions = await db.studentRestriction.find({
```

Comment:
Please restrict the selection and adjust the order to comply with other queries where the relations go after the selection.
```ts
 const studentRestriction = await db.studentRestriction.findOne({
  select: {
    id: true,
    restriction: { restrictionCode: true },
  },
  relations: { restriction: true },
  where: {
    student: { id: student.id },
    restriction: { restrictionCode: RestrictionCode.E2 },
  },
});
```